packages/terra-base/src/Base.jsx

```diff
@@ -1,38 +1,75 @@
 import React, { PropTypes } from 'react';
-import './Base.scss';
+
+import { I18nProvider, i18nLoader } from 'terra-i18n';
+
+import BaseStyles from './BaseStyles';
 
 const propTypes = {
   /**
    * The component(s) that will be wrapped by `<Base />`.
    */
   children: PropTypes.node.isRequired,
+  /**
+   * The locale name.
+   */
+  locale: PropTypes.string.isRequired,
+  /**
+   * Customized translations provided by consuming application
+   * only for current locale.
+   */
+  customMessages: PropTypes.object,
+};
+
+const defaultProps = {
+  customMessages: {},
 };
 
 class Base extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      load: false,
+      locale: props.locale,
+      messages: {},
+    };
+  }
 
   componentDidMount() {
-    // Checks to run when not in production
-    if (process.env.NODE_ENV !== 'production') {
-      // Check to ensure terra-Base class is set on html element
-      if (!new RegExp('(^|\\s)terra-Base(\\s|$)').test(document.documentElement.className)) {
-        // eslint-disable-next-line
-        console.warn('The html element is missing the terra-Base class.');
-      }
-
-      // Check to ensure dir attribute is set on html element
-      if (!document.documentElement.hasAttribute('dir')) {
-        // eslint-disable-next-line
-        console.warn('The html element is missing the dir attribute. For terra directionality based styles to render correctly, add dir="ltr" or dir="rtl" to the html element.');
-      }
-    }
+    i18nLoader(this.props.locale, this.setState, this);
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (this.props === nextProps) return;
+    i18nLoader(nextProps.locale, this.setState, this);
   }
 
   render() {
-    const { children, ...customProps } = this.props;
-    return <div {...customProps}>{children}</div>;
+    if (!this.state.load) return null;
+
+    const {
+      children,
+      locale,
+      customMessages,
+      ...customProps
+    } = this.props;
+
+    const messages = Object.assign({}, this.state.messages, customMessages);
+
+    return (
+      <I18nProvider
+        locale={this.state.locale}
```

Comment:
Just so I understand what's happening here, does the messages object only contain the messages for the given locale?  And if so what is the purpose of locale?  Is it to format things like numbers, dates, etc.?